gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -1603,6 +1603,35 @@ def raw_file_to_dataset(self, file_path: str) -> xr.Dataset:
         else:
             raise ValueError('Expected either "file" or "s3" protocol')
 
+    def convert_raw_times_to_comparable_times(self, ds: xr.Dataset, time_dim: str) -> np.ndarray | list:
```

Comment:
Ideally, the type hint would specify one of the two current possible return types, since the two have very different methods that can be called on them. Is there any way this function can return a `list`? My hunch is it can only return `np.ndarray`

Reply:
The downstream implementation returns a list but I've changed that in the responsible manager so now we can verifiably return a `np.ndarray`. Better that way.